src/nemo_evaluator/observability/model_traffic.py

```
@@ -132,6 +133,52 @@ def _truncate(value: Any, limit: int) -> Any:
     return value
 
 
+def _first_str(data: dict[str, Any], *keys: str) -> str:
+    for key in keys:
+        value = data.get(key)
+        if value is not None and value != "":
+            return str(value)
+    return ""
+
+
+def _error_summary(body: Any, max_content_chars: int) -> dict[str, Any]:
```

Ruff (0.15.18)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `body`

(ANN401)